### PR TITLE
UNOMI-294 Implement createOrUpdatePersona mutation

### PR DESCRIPTION
**Pull request by pmi**
- Added score to CDPInterest constructor
- Refactored GraphQlSchemaUpdater to better fit for multiple dynamic types

### CODE REVIEW

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/commands/CreateOrUpdatePersonaCommand.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            Map<String, Double> interestMap = new HashMap<>();
+            personaInput.getCdp_interests().forEach(interestInput -> {
+                interestMap.put(interestInput.getTopic(), interestInput.getScore());
+            });
+            persona.getProperties().put("interests", interestMap);
+        }
+
+        personaProps.forEach(propertyType -> {
+            final String propId = propertyType.getItemId();
+            persona.setProperty(propId, personaAsMap.get(propId));
```
Comment (anatol-sialitski) [inline graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/commands/CreateOrUpdatePersonaCommand.java]: Please use `PropertyNameTranslator.translateFromGraphQLToUnomi(propertyName)`

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/commands/CreateOrUpdatePersonaCommand.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.unomi.graphql.commands;
+
+import org.apache.unomi.api.Consent;
```
Comment (anatol-sialitski) [inline graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/commands/CreateOrUpdatePersonaCommand.java:19]: Please optimize imports

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CustomerPropertyDataFetcher.java [hunk 3]
```diff
@@ -19,21 +19,25 @@
 import graphql.schema.DataFetchingEnvironment;
 import org.apache.unomi.graphql.schema.PropertyNameTranslator;
 import org.apache.unomi.graphql.types.output.CDPEvent;
+import org.apache.unomi.graphql.types.output.CDPPersona;
 import org.apache.unomi.graphql.types.output.CDPProfile;
 
-public class CustomerPropertyDataFetcher extends BaseDataFetcher<Object> {
+public class CustomerPropertyDataFetcher extends DynamicFieldDataFetcher<Object> {
 
     private final String propertyName;
```
Comment (anatol-sialitski) [inline graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CustomerPropertyDataFetcher.java]: please remove that property

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CDPProfilePropertiesFilterDataFetcher.java [hunk 4]
```diff
@@ -30,16 +29,14 @@
...
-    private final String fieldName;
+public class CDPProfilePropertiesFilterDataFetcher extends DynamicFieldDataFetcher<Object> {
 
     private final String propertyName;
 
     private final String comparisonOperator;
 
     public CDPProfilePropertiesFilterDataFetcher(final String fieldName) {
-        this.fieldName = fieldName;
+        super(fieldName);
```
Comment (anatol-sialitski) [inline graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CDPProfilePropertiesFilterDataFetcher.java:39]: `PropertyNameTranslator.translateFromGraphQLToUnomi(fieldName)`

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/DynamicFieldDataFetcher.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+package org.apache.unomi.graphql.fetchers;
+
+import graphql.schema.DataFetcher;
+
+public abstract class DynamicFieldDataFetcher<T> implements DataFetcher<T> {
+
+    protected String fieldName;
+
+    public DynamicFieldDataFetcher(String fieldName) {
+        this.fieldName = fieldName;
```
Comment (anatol-sialitski) [inline graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/DynamicFieldDataFetcher.java]: please use `PropertyNameTranslator.translateFromGraphQLToUnomi(propertyName)`

File: graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CustomerPropertyDataFetcher.java (hunk 3 above)
Comment (pmi) [reply graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/fetchers/CustomerPropertyDataFetcher.java:27]: No! Default constructor of DynamicFieldDataFetcher should always accept field name as string.

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**